test/test_EKF_basics.cpp

```diff
@@ -34,128 +34,33 @@
 #include <gtest/gtest.h>
 #include <math.h>
 #include "EKF/ekf.h"
+#include "SensorSimulator/SensorSimulator.h"
+
 
 class EkfInitializationTest : public ::testing::Test {
  public:
 
-	Ekf _ekf{};
-
-	// Basics sensors
-	const uint32_t _imu_dt_us{4000};	// 250 Hz	Period between IMU updates
-	const uint32_t _baro_dt_us{12500};	// 80 Hz	Period between barometer updates
-	const uint32_t _mag_dt_us{12500};	// 80 Hz	Period between magnetometer updates
-	const uint32_t _gps_dt_us{200000};	// 5 Hz		Period between GPS updates
-
-	// Flags that control if a sensor is fused
-	bool _fuse_imu{true};
-	bool _fuse_baro{true};
-	bool _fuse_mag{true};
-	bool _fuse_gps{false};	// GPS measurements are expected to not come in from beginning
-
-	// GPS message
-	gps_message _gps_message{};
-
-	uint32_t _update_dt_us{};			// greatest common divider of all basic sensor periods
-	const uint32_t _init_duration_us{2000000};	// 2s	Duration of
+	Ekf* _ekf;
+	SensorSimulator* _sensor_simulator;
```

Comment:
I don't understand why you're using pointers here. Can't you just create an instance of EKF and SensorSimulator? Then you don't have to manage the heap memory and worry about leaking it in case of a crash. I see the teardown, but you never know what happens in between